src/providers/info.rs

```diff
@@ -76,22 +76,22 @@ impl DefaultInfoProvider<'_> {
 
     fn cached<T>(
         &mut self,
-        local_cache: impl Fn(&mut Self) -> &mut HashMap<String, T>,
+        local_cache: impl Fn(&mut Self) -> &mut HashMap<String, Option<T>>,
         url: impl Into<String>,
     ) -> Result<T>
         where
             T: Clone + DeserializeOwned,
     {
         let url = url.into();
         match local_cache(self).get(&url) {
-            Some(value) => Ok(value.clone()),
+            Some(value) => value.clone().ok_or(anyhow::anyhow!("Could not decode cached")),
             None => {
                 let data =
                     self.cache
                         .request_cached(self.client, &url, info_cache_duration())?;
-                let value: T = serde_json::from_str(&data)?;
+                let value: Option<T> = serde_json::from_str(&data).unwrap_or(None);
```

Comment:
Aren't we drowning this panic here? Maybe we can use `unwrap_or_else` and throw a message in `dbg!` if we fail?

Reply:
We are expecting a panic here, therefore I would not lock it ... this should be refactored in the future (that is what I added to #54 )